File: android/app/src/main/java/com/hedvig/app/NativeRoutingPackage.java
package com.hedvig.app;

import com.apollographql.apollo.ApolloClient;
import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class NativeRoutingPackage implements ReactPackage {

    private final ApolloClient apolloClient;

    NativeRoutingPackage(ApolloClient apolloClient) {
        this.apolloClient = apolloClient;
    }

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        return Arrays.asList(new NativeRoutingModule(reactContext, apolloClient));
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: android/app/src/main/java/com/hedvig/app/NativeRoutingPackage.kt
package com.hedvig.app

import com.apollographql.apollo.ApolloClient
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

import java.util.Arrays
import java.util.Collections

class NativeRoutingPackage internal constructor(private val apolloClient: ApolloClient) : ReactPackage {

    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return Arrays.asList<NativeModule>(NativeRoutingModule(reactContext, apolloClient))
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }
}
